### DISCUSSION TITLE
What's the meaning of CSL

### DISCUSSION THREAD

**Discussion by Maxppddcsz** [NVIDIA/open-gpu-kernel-modules#562] [Q&A] [answered]
When I was reading the source code related to UVM memory, I found a term “CSL context”. Could you please tell me what CSL stands for? I have searched many documents but haven’t been able to understand its meaning. Thank you very much.
`/*******************************************************************************
    CSL Interface and Locking

    The following functions do not acquire the RM API or GPU locks and must not be called
    concurrently with the same UvmCslContext parameter in different threads. The caller must
    guarantee this exclusion.

    * nvUvmInterfaceCslRotateIv
    * nvUvmInterfaceCslEncrypt
    * nvUvmInterfaceCslDecrypt
    * nvUvmInterfaceCslSign
    * nvUvmInterfaceCslQueryMessagePool
    * nvUvmInterfaceCslIncrementIv
*/`

**Comment by johnhubbard** [comment, marked as answer]
CSL stands for Cryptographic Services Library.

**Reply by steven-bellock** [reply]
We'll add a definition of CSL in `nv_uvm_interface.h`.

**Reply by Maxppddcsz** [reply]
Thank you very much for your timely reply!

**Reply by VicoinDub** [reply]
So what is CCSL? I found a lot of code about CCSL

**Reply by steven-bellock** [reply]
@VicoinDub It is a wrapper around the Linux Kernel Crypto API that provides cryptography services to `nvidia_uvm.ko` and `nvidia.ko` when the system is in confidential compute mode.

**Comment by VicoinDub** [comment]
Thank you very much!



---Original---
From: "Steven ***@***.***&gt;
Date: Fri, Sep 27, 2024 22:11 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [NVIDIA/open-gpu-kernel-modules] What's the meaning of CSL(Discussion #562)




 
@VicoinDub It is a wrapper around the Linux Kernel Crypto API that provides cryptography services to nvidia_uvm.ko and nvidia.ko when the system is in confidential compute mode.
